numpy/typing/tests/data/pass/ufunclike.py

```diff
@@ -13,8 +13,8 @@ def __floor__(self) -> Object:
     def __ge__(self, value: object) -> bool:
         return True
 
-    def __array__(self, dtype: Optional[np.typing.DTypeLike] = None,
-                  copy: Optional[bool] = None) -> np.ndarray[Any, np.dtype[np.object_]]:
+    def __array__(self, dtype: np.typing.DTypeLike = None,
```

Comment:
I would leave the `| None` here, it has a different meaning (and there is an open PR to deprecate `None`)
```suggestion
    def __array__(self, dtype: np.typing.DTypeLike | None = None,
```

Reply:
That's not need, as it's already included within `DTypeLike`: https://github.com/numpy/numpy/blob/v2.0.0/numpy/_typing/_dtype_like.py#L118-L129

Reply:
Which PR is that? I can't seem to find it 🤔